sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/operation_status.hpp"
+#include "azure/core/response.hpp"
+
+#include <chrono>
+#include <string>
+#include <memory>
+
+namespace Azure { namespace Core {
+
+  //@param <T> The type of the final result of the long-running operation.
+  template <class T> class Operation {
+  private:
+    virtual std::unique_ptr<Http::RawResponse> PollInternal(Context& context) = 0;
+    virtual Response<T> PollUntilDoneInternal(std::chrono::milliseconds period, Context& context)
+        = 0;
+
+  protected:
+    OperationStatus m_status = OperationStatus::NotStarted;
+
+  public:
+    OperationStatus Status() const { return m_status; }
+
+    bool Done() const
```

Comment:
```suggestion
    bool IsDone() const
```

Reply:
I'm OK with IsDone.

Reply:
There is no consistency across languages on the Operation type.  Every language did their own thing.  IsDone is close to what what JS used.  I'm ok with that.  I'll make a change

Reply:
For context, my suggestion was less about consistency with other languages and more about what I would consider a naming convention for bool returning "test" methods. I, as the caller, am asking the API a question, and it returns a yes/no bool answer, so the name shouldn't just be a verb or noun but rather a query on the verb, like `IsFinished, or `HasValue`, etc.